default/AI/Behavior/Character.py

```diff
@@ -0,0 +1,305 @@
+        """Select a fraction less than 1 at maximum ratio of PP for buildings"""
+        return None
+
+    def preferred_discount_multiplier(self, alternatives):  # pylint: disable=no-self-use,unused-argument
+        """Select a discount multiplier for use in evaluate planet"""
+        return None
+
+    def max_defense_portion(self):  # pylint: disable=no-self-use,unused-argument
+        """Return maximum fraction of production for defense"""
+        return 1.0
+
+    def check_orbital_production(self):  # pylint: disable=no-self-use,unused-argument
+        """Return if orbital production is checked this turn"""
+        return False
+
+    def target_number_of_orbitals(self):  # pylint: disable=no-self-use,unused-argument
+        """Return target number of orbitals"""
+        return 0
+
+    def may_build_building(self, building):  # pylint: disable=no-self-use,unused-argument
+        """Return True if permitted to build ''building''"""
+        return True
+
+    def may_produce_troops(self):  # pylint: disable=no-self-use,unused-argument
+        """Return True if permitted to produce troop ships"""
+        return True
+
+    def military_safety_factor(self):  # pylint: disable=no-self-use,unused-argument
+        """Return military safety factor"""
+        return 0.0
+
+    def may_dither_focus_to_gain_research(self):  # pylint: disable=no-self-use,unused-argument
+        """Return True if permitted to trade production at a loss for research"""
+        return True
+
+    def may_research_heavily(self):  # pylint: disable=no-self-use,unused-argument
+        """Return true if allowed to target research/industry > 1.5"""
+        return True
+
+    def may_travel_beyond_supply(self, distance):  # pylint: disable=no-self-use,unused-argument
+        """Return True if able to travel distance hops beyond empire supply"""
+        return True
+
+    def get_research_index(self):  # pylint: disable=no-self-use,unused-argument
+        """Deprecated"""
+        return None
+
+    def may_research_xeno_genetics_variances(self):  # pylint: disable=no-self-use,unused-argument
+        """Return True if able to vary xeno genetics research"""
+        return True
+
+    def prefer_research_defensive(self):  # pylint: disable=no-self-use,unused-argument
+        """Return True if should prefer defensive tech research"""
+        return True
+
+    def prefer_research_low_aggression(self):  # pylint: disable=no-self-use,unused-argument
+        """Return True if should prefer less aggressive tech research"""
+        return True
+
+    def may_research_tech(self, tech):  # pylint: disable=no-self-use,unused-argument
+        """Return True if permitted to research ''tech''"""
+        return True
+
+    def may_research_tech_classic(self, tech):  # pylint: disable=no-self-use,unused-argument
+        """Return True if permitted to research ''tech'' in the classic algorithm"""
+        return True
+
+
+class Character(Behavior):
```

Comment:
It is a collection of Behaviors that implements the interface of Behavior.  It is both.

I chose not to use abstract base class (abc) for the inheritance because that forces you to re-implement every method.  Most Behaviors should only need to re-implement a few methods.  Re implementing all of them adds noise, hides the purpose of the Behavior and can only introduce bugs.